src/main/java/com/iexec/worker/sgx/SgxService.java

```diff
@@ -62,17 +61,18 @@ public SgxService(
 
     @PostConstruct
     void init() {
-        boolean sgxSupported = isSgxSupported(sgxDriverMode);
-        final boolean driverModeNotNone = SgxDriverMode.isDriverModeNotNone(sgxDriverMode);
-        if (!sgxSupported && driverModeNotNone) {
-            this.sgxEnabled = false;
-            log.error("SGX required but not supported by worker. Shutting down. " +
-                    "[sgxDriverMode: {}]", sgxDriverMode);
-            SpringApplication.exit(context, () -> 1);
+        sgxEnabled = SgxDriverMode.isDriverModeNotNone(sgxDriverMode);
+        if (!sgxEnabled) {
+            log.info("No SGX driver defined, skipping SGX check [sgxDriverMode:{}]", sgxDriverMode);
             return;
         }
-
-        this.sgxEnabled = driverModeNotNone;
+        // sgxEnabled is always true when reaching this line
+        // sgxEnabled && isSgxSupported can be simplified to isSgxSupported
```

Comment:
I'm not sure these comments adds anything. They refer to some code that will not be there anymore. They may be considered as misleading. Don't you think so?

Reply:
would it be better with only
```suggestion
        // sgxEnabled is always true when reaching this line
        // sgxEnabled becomes equal to isSgxSupported in this case
```

Reply:
?

Reply:
```suggestion
        // SgxDriverMode.isDriverModeNotNone is always true when reaching this line
        // sgxEnabled becomes equal to isSgxSupported in this case
```